include/deal.II/base/mpi_compute_index_owner_internal.h

```diff
@@ -114,15 +116,19 @@ namespace Utilities
                 for (types::global_dof_index i = interval.first;
                      i < interval.second;
                      i++)
-                  Assert(actually_owning_ranks[i - local_range.first] ==
-                           numbers::invalid_unsigned_int,
-                         ExcInternalError());
-                Assert(interval.first >= local_range.first &&
-                         interval.first < local_range.second,
-                       ExcInternalError());
-                Assert(interval.second > local_range.first &&
-                         interval.second <= local_range.second,
-                       ExcInternalError());
+                  Assert(
+                    actually_owning_ranks[i - local_range.first] ==
+                      numbers::invalid_unsigned_int,
+                    ExcMessage(
+                      "Multiple processes seem to own the same global index. "
+                      "A possible reason is that the sets of locally owned "
+                      "indices are not distinct."));
+                Assert(interval.first < interval.second, ExcInternalError());
```

Comment:
We should move this assertion outside of the loop - it can never trigger in it since this is also the loop bounds.

Reply:
Oh, nevermind - there are no brackets around the first assert so this is outside the loop!